src/main/java/uk/gov/hmcts/ccd/domain/service/common/StateReferenceService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.casestate.EnablingConditionParser;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Service
+public class StateReferenceService {
+
+    private final EnablingConditionParser enablingConditionParser;
+
+    @Inject
+    public StateReferenceService(EnablingConditionParser enablingConditionParser) {
+        this.enablingConditionParser = enablingConditionParser;
+    }
+
+    public Optional<String> evaluatePostStateCondition(List<EventPostStateDefinition> eventPostStateDefinitions,
+                                                       Map<String, JsonNode> caseEventData) {
+        List<EventPostStateDefinition> postStateDefinitions = Optional
+            .ofNullable(eventPostStateDefinitions)
+            .orElse(new ArrayList<>());
+
+        for (EventPostStateDefinition eventPostStateDefinition : postStateDefinitions) {
+            String enablingCondition = eventPostStateDefinition.getEnablingCondition();
+            if (enablingCondition != null) {
```

Comment:
I would add a method `isDefault()` to EventPostStateDefinition which itself checks whether enablingCondition != null. This way the rule about when a condition is a default condition is encapsulated in one place rather than scattered around

Reply:
isDefault() method added to EventPostStateDefinition